third_party/zap/templates/README.md

```diff
@@ -0,0 +1,26 @@
+# ZAP generation templates
+
```

Comment:
Since the templates here are only used to generate the `gen/` folder of examples, what about moving them out of this PR and do a separate PRs to land them into `examples/zap-templates` instead of `third_party/zap` ?

Basically I would like to get this PR to a point where the focus is on the changes on the `gen/` files and the side effect on the `src/app` code because of what it induces.

Reply:
See PR #3638 . Templates should still be directly under `third_party/zap` since they're dependent of ZAP. Moving them elsewhere would give the impression that they can be "standalone" which they are not